tests/algorithms/pytorch/test_newton_raphson.py

```diff
@@ -22,6 +22,17 @@
 EXPECTED_PERFORMANCE = 0
 
 
+@pytest.fixture
+def x_y_dataset(numpy_torch_dataset):
+    class XYDataset(numpy_torch_dataset):
+        def __init__(self, datasamples, is_inference=True) -> None:
+            self.x = datasamples[0]
+            self.y = datasamples[1]
+            self.is_inference = is_inference
+
+    return XYDataset
+
+
```

Comment:
I think it is not necessary to create a new intermediate fixture, the `numpy_torch_dataset` format can be updated directly

Reply:
done